### PR TITLE
Fix TextTestResult and TextTestRunner in unittest

### PR DESCRIPTION
**Pull request by c-bata**
It looks `stream` property is `IO[str]` type, not `TextIO` because it calls `write` and other methods. So I changes it.

And I also add types for other attributes in `TextTestResult` and `TextTestRunner`.

### CODE REVIEW

File: stdlib/3/unittest/__init__.pyi [hunk 1]
```diff
@@ -298,22 +302,30 @@ _TextTestResult = TextTestResult
 
 defaultTestLoader = ...  # type: TestLoader
 
-_ResultClassType = Callable[[TextIO, bool, int], TestResult]
+_ResultClassType = Callable[[IO[str], bool, int], TestResult]
 
 class TestRunner:
     def run(self, test: Union[TestSuite, TestCase]) -> TestResult: ...
 
 class TextTestRunner(TestRunner):
```
Comment (c-bata) [inline stdlib/3/unittest/__init__.pyi]: The source code of cpython is here.
https://github.com/python/cpython/blob/0a16bb15afec28f355bc28203b6b10610293f026/Lib/unittest/runner.py#L120-L221

File: stdlib/3/unittest/__init__.pyi [hunk 2]
```diff
@@ -289,7 +289,11 @@ class TestResult:
 class TextTestResult(TestResult):
```
Comment (c-bata) [inline stdlib/3/unittest/__init__.pyi]: The source code of cpython is here.
https://github.com/python/cpython/blob/0a16bb15afec28f355bc28203b6b10610293f026/Lib/unittest/runner.py#L29-L117

File: stdlib/3/unittest/__init__.pyi [hunk 3]
```diff
@@ -298,22 +302,30 @@ _TextTestResult = TextTestResult
...
 
 class TextTestRunner(TestRunner):
+    stream: IO[str]
+    descriptions: bool
+    verbosity: int
+    failfast: bool
+    buffer: bool
+    tb_locals: bool
+    warnings: Optional[Type[Warning]]
+    resultclass: Optional[_ResultClassType]
```
Comment (srittau) [inline stdlib/3/unittest/__init__.pyi]:
```suggestion
    resultclass: _ResultClassType
```

This is initialized to `TextTestResult` and not overwritten if `resultclass` is `None`.

**Comment by c-bata** [comment]
Hi @srittau. Thanks for reviewing. I've just added your suggested change.

**Pull request merged**